include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -538,6 +561,141 @@ make_transform_iterator(_Iter __it, _UnaryFunc __unary_func)
     return transform_iterator<_Iter, _UnaryFunc>(__it, __unary_func);
 }
 
+template <typename _Iter, typename _UnaryFunc>
+class transform_output_iterator
+{
+  private:
+    _Iter __my_it_;
+    const _UnaryFunc __my_unary_func_;
```

Comment:
Why this field is const? Is it really required?

Reply:
The field is const to require `operator()` of the functor to be a const method, preventing mutation of any state in the functor when it's invoked on element read I think.

Reply:
I marked this field as const because once the unary function is initialized once in the constructor, it is no longer modified elsewhere. Is there a reason to change this?